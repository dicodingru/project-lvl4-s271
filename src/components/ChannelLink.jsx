import React from 'react';
import PropTypes from 'prop-types';

const ChannelLink = (props) => {
  const { name, isActive, onClick } = props;
  const linkClass = `list-group-item${isActive ? ' active' : ''}`;
  return (
    <a className={linkClass} href={`#${name}`} onClick={onClick}>
      <span>{name}</span>
    </a>
  );
};

ChannelLink.propTypes = {
  name: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};

export default ChannelLink;
