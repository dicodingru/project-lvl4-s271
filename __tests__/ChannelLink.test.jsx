import React from 'react';
import renderer from 'react-test-renderer';
import ChannelLink from '../src/components/ChannelLink';

describe('ChannelLink component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<ChannelLink name="general" />);
  });

  test('should render successfully', () => {
    expect(component.exists()).toEqual(true);
  });

  test('should match snapshot', () => {
    component = renderer.create(<ChannelLink name="general" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
