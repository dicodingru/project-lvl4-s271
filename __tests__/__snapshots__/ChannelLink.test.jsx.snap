// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelLink component should match snapshot 1`] = `
<div
  className="sc-bdVaJa jWKkjw"
>
  <a
    href="#general"
  >
    <span>
      general
    </span>
  </a>
</div>
`;
